### PR TITLE
Update admin client command line interface to based on AdminAPI

### LINKED ISSUE #155: Clean up admin client API
- Some codes are unused 
- Refactor command line admin client to based on the AdminAPI class

### PR DESCRIPTION
**Pull request by YuanTingHsieh**
Fix #155

- Remove unused codes
- Refactor command line admin client (lib.py -> cli.py) to based on the AdminAPI class
- Introduce an AdminAPISpec base class

### CODE REVIEW

**Comment by nvkevlu** [comment]
Thank you @YuanTingHsieh, the changes look very good for the most part. I wanted to test fully to make sure everything still works, and there is an issue with previous fl_admin.py scripts in an SSL configuration not working because poc is always set to True.

**Review by nvkevlu — CHANGES_REQUESTED** [review]

File: nvflare/fuel/hci/client/cli.py [hunk 1]
```diff
@@ -161,62 +60,61 @@ class AdminClient(cmd.Cmd):
...
+        self.api = AdminAPI(
+            host=host,
+            port=port,
+            ca_cert=ca_cert,
+            client_cert=client_cert,
+            client_key=client_key,
+            server_cn=server_cn,
+            cmd_modules=modules,
+            debug=False,
+            poc=True,
```
[the remark this answers is not in the document]
Comment (YuanTingHsieh) [reply nvflare/fuel/hci/client/cli.py]: thanks, fixed.

**Review by nvkevlu — APPROVED** [review]
Thanks, this now works with all the setups I tried.

**Review by yanchengnv — APPROVED** [review]

File: nvflare/fuel/hci/client/cli.py [hunk 2]
```diff
@@ -161,62 +60,61 @@ class AdminClient(cmd.Cmd):
...
         ca_cert=None,
         client_cert=None,
         client_key=None,
         server_cn=None,
-        require_login=False,
-        credential_type="password",
-        cmd_modules=None,
-        debug=False,
+        require_login: bool = False,
+        credential_type: Union[str, CredentialType] = CredentialType.PASSWORD,
```
[the remark this answers is not in the document]
Comment (YuanTingHsieh) [reply nvflare/fuel/hci/client/cli.py:79]: good point, let me submit a small fix PR follow this up.

[the remark this answers is not in the document]
Comment (YuanTingHsieh) [reply nvflare/fuel/hci/client/cli.py:79]: fixed in #188

**Pull request merged**